code/modules/hydroponics/beekeeping/smoker.dm

```diff
@@ -23,10 +23,12 @@
 
 //I would prefer to rename this to attack(), but that would involve touching hundreds of files.
 /obj/item/weapon/bee_smoker/resolve_attackby(atom/A, mob/user)
-	if (istype(A, /obj/structure/reagent_dispensers/fueltank) && get_dist(src,A) <= 1)
-		A.reagents.trans_to_obj(src, max_fuel)
-		to_chat(user, "<span class='notice'>Smoker refilled!</span>")
-		playsound(src.loc, 'sound/effects/refill.ogg', 50, 1, -6)
+	if (istype(A, /obj/item/weapon/reagent_containers/glass))
```

Comment:
It would make more sense to have you attack the smoker with a container, instead of attacking the container with the smoker.

Reply:
Resolved.